### PR TITLE
[Trivial] Remove dead code

### PR DESCRIPTION
**Pull request by thejohnfreeman**
What it says on the tin. Unused methods, undefined declarations, unused members, etc.

### CODE REVIEW

File: src/ripple/overlay/impl/OverlayImpl.cpp [hunk 1]
```diff
@@ -99,9 +99,7 @@ OverlayImpl::Timer::on_timer(error_code ec)
     overlay_.m_peerFinder->once_per_second();
     overlay_.sendEndpoints();
     overlay_.autoConnect();
-
-    if ((overlay_.timer_count_ % Tuning::checkIdlePeers) == 0)
```
Comment (gregtatcam) [inline src/ripple/overlay/impl/OverlayImpl.cpp:103]: Why is this deleted? This was added so that we don't check for idle peers too frequently.

Comment (thejohnfreeman) [reply src/ripple/overlay/impl/OverlayImpl.cpp:103]: `timer_count_` is [initialized](https://github.com/ripple/rippled/pull/3753/files#diff-3f3119d499efbf4f0ec803cf22831e82d48df5a396e321bb343961c920454b8eL141) to 0 and never updated. It is always 0, and this condition is always true.

Comment (gregtatcam) [reply src/ripple/overlay/impl/OverlayImpl.cpp:103]: oh, right! it's a bug then! it should be incremented.

Comment (gregtatcam) [reply src/ripple/overlay/impl/OverlayImpl.cpp:103]: i see what happened. it used to be this:

    if ((++overlay_.timer_count_ % Tuning::checkSeconds) == 0)
        overlay_.check();

    if ((overlay_.timer_count_ % Tuning::checkIdlePeers) == 0)
        overlay_.deleteIdlePeers();

but then two top lines were deleted in one of the unrelated commits.

File: src/ripple/nodestore/impl/Shard.cpp [hunk 2]
```diff
@@ -583,17 +583,6 @@ Shard::finalize(
...
-        /*
-        TODO MP
-        A lock is required when calling the NuDB verify function. Because
-        this can be a time consuming process, the server may desync.
-        Until this function is modified to work on an open database, we
-        are unable to use it from rippled.
-
-        // Verify backend integrity
-        backend_->verify();
-        */
```
Comment (seelabs) [inline src/ripple/nodestore/impl/Shard.cpp:595]: Let's check with @miguelportilla before removing this and the `verify` implementation.

File: src/ripple/overlay/impl/OverlayImpl.cpp (hunk 1 above)
Comment (seelabs) [reply src/ripple/overlay/impl/OverlayImpl.cpp:103]: This doesn't seem like the right PR to fix that bug, but I wouldn't remove the code either. Let's just keep the code for now and open an issue so we don't lose track of this. @gregtatcam can you open an issue?

**Review by seelabs — COMMENTED** [review]
This looks good - thanks for the cleanup. Left two minor comments.

File: src/ripple/overlay/impl/OverlayImpl.cpp (hunk 1 above)
Comment (gregtatcam) [reply src/ripple/overlay/impl/OverlayImpl.cpp:103]: I opened the issue #3754.

File: src/ripple/nodestore/impl/Shard.cpp (hunk 2 above)
Comment (miguelportilla) [reply src/ripple/nodestore/impl/Shard.cpp:595]: I am okay removing this commented out code. Although the backend verify function isn't used, it could be useful at some point. For instance, the above situation can be addressed by calling verify on the new deterministic backend. I could also see a server startup switch to have the backend be verified at startup.

**Review by cryptobrad — COMMENTED** [review]
:100: 

There is some other functions or things that are dead too

`AcceptedLedger::getTxn(int)` at https://github.com/ripple/rippled/blob/develop/src/ripple/app/ledger/AcceptedLedger.h#L72-L73

`AutoSocket::{setSSLOnly|setPlayOnly}` at https://github.com/ripple/rippled/blob/develop/src/ripple/net/AutoSocket.h#L80-L89

`LedgerReplayTaks::getCountedObjectName` at https://github.com/ripple/rippled/blob/develop/src/ripple/app/ledger/LedgerReplayTask.h#L132-L136

`LedgerMaster::getBuildingLedger` at https://github.com/ripple/rippled/blob/develop/src/ripple/app/ledger/impl/LedgerMaster.cpp#L572-L577

`LedgerMaster::NO_VALIDATED_LEDGER_AGE` at https://github.com/ripple/rippled/blob/develop/src/ripple/app/ledger/LedgerMaster.h#L76-L77

`NetworkOps::{getTxMap|hasTxSet}` at https://github.com/ripple/rippled/blob/develop/src/ripple/app/misc/NetworkOPs.cpp#L366-L372

and also these: 

https://github.com/ripple/rippled/blob/develop/src/ripple/app/misc/NetworkOPs.cpp#L144-L145

`NetworksOPs::getConsensusLCL` at https://github.com/ripple/rippled/blob/develop/src/ripple/app/misc/NetworkOPs.h#L244-L245

`LedgerDeltaAcquire::getCountedObjectName` at https://github.com/ripple/rippled/blob/develop/src/ripple/app/ledger/impl/LedgerDeltaAcquire.h#L114-L118

`LedgerReplayTaks::getCountedObjectName` at https://github.com/ripple/rippled/blob/develop/src/ripple/app/ledger/LedgerReplayTask.h#L132-L136

**Comment by pwang200** [comment]
As @cryptobrad said `LedgerDeltaAcquire::getCountedObjectName` and `
LedgerReplayTaks::getCountedObjectName` should be removed. They are not required by `CountedObject` any more.   `SkipListAcquire::getCountedObjectName`  https://github.com/ripple/rippled/blob/80bd107e570b9112de3b2ef52937960986ad5347/src/ripple/app/ledger/impl/SkipListAcquire.h#L112-L116 should also be removed.

**Review by seelabs — APPROVED** [review]
:+1: Nice cleanups

**Comment by thejohnfreeman** [comment]
Thanks for the suggestions! I'll work to include them, and another I identified today, so I'll put the PR in draft mode temporarily.

**Review by gregtatcam — COMMENTED** [review]
Also unused OverlayImpl::serverHandler() and PeerImp::networkID()

**Review by nbougalis — APPROVED** [review]
